compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Utils/AMDAIEDmaUtils.h

```diff
@@ -120,14 +120,39 @@ LogicalResult combineAccessPatterns(RewriterBase &rewriter,
 /// number of max sizes might exceed the number of sizes and the other way
 /// around, BUT after canonicalization, the number of sizes should be smaller or
 /// equal to the number of max sizes (if specified).
-LogicalResult foldLinearDims(MLIRContext *ctx,
-                             const SmallVector<OpFoldResult> &offsets,
-                             const SmallVector<OpFoldResult> &sizes,
-                             const SmallVector<OpFoldResult> &strides,
-                             SmallVector<OpFoldResult> &newOffsets,
-                             SmallVector<OpFoldResult> &newSizes,
-                             SmallVector<OpFoldResult> &newStrides,
-                             ArrayRef<int64_t> maxSizes = {});
+LogicalResult foldLinearDims(
+    MLIRContext *ctx, const SmallVector<OpFoldResult> &offsets,
+    const SmallVector<OpFoldResult> &sizes,
+    const SmallVector<OpFoldResult> &strides,
+    SmallVector<OpFoldResult> &newOffsets, SmallVector<OpFoldResult> &newSizes,
+    SmallVector<OpFoldResult> &newStrides,
+    function_ref<bool(size_t, int64_t)> checkValidSize =
+        [](size_t idxFromEnd, int64_t size) { return true; });
+
+/// Utility to fold a provided repetition count from the front of the access
+/// pattern (dimensions with `size > 1` and `stride == 0` indicate a
+/// repetition). This function fails if the access pattern doesn't have the
+/// provided repetition count at the front.
+///
+/// This transformation is useful for circular DMA operations that don't need
+/// the repetition count to be specified. Note however that not all repetition
+/// dimensions are necessarily completely removed as that could potentially
+/// invalidate the DMA operation as part of the repetition could be operating on
+/// the same data slice, while part could be operating on a new data slice. This
+/// function doesn't make any assumption on that, but that's the reason why an
+/// optional repetition count can be passed.
+///
+/// Example:
+///
+/// sizes: [32, 8], strides: [0, 1], repetitionCount: 16
+///
+/// will be transformed into:
+///
+/// sizes: [2, 8], strides: [0, 1]
+LogicalResult foldRepetitionCount(
+    MLIRContext *ctx, SmallVector<OpFoldResult> &sizes,
+    SmallVector<OpFoldResult> &strides,
+    std::optional<int64_t> maybeRepetitionCount = std::nullopt);
```

Comment:
I found that the only place where the optional `maybeRepetitionCount` is provided is in `AMDAIEDmaUtilsTest.cpp`. Are there other parts of the code where it is actually needed?

Reply:
Yeah, I am staging a couple of changes through a few PR and this will be needed in a follow-up, see: https://github.com/nod-ai/iree-amd-aie/compare/main...jtuyls:iree-amd-aie:increase-l2-tile-sizes#diff-888fdafbbd093dc98e4fae55a26e7fa56065fc7506fedc0107b8a897c19f816aR212